shell/common/shell.cc

```diff
@@ -66,8 +78,8 @@ base::LazyInstance<NonDiscardableMemoryAllocator> g_discardable;
 void ServiceIsolateHook(bool running_precompiled) {
   if (!running_precompiled) {
     const blink::Settings& settings = blink::Settings::Get();
-    if (settings.enable_observatory)
-      DiagnosticServer::Start();
+    if (settings.enable_observatory && settings.enable_diagnostic)
```

Comment:
This was preexisting behavior - I think the assumption was that anyone who disabled the observatory wouldn't want the diagnostic server either.